overlord/devicestate/systems.go

```diff
@@ -83,28 +83,39 @@ func systemFromSeed(label string) (*System, error) {
 	if err != nil {
 		return nil, fmt.Errorf("cannot obtain brand: %v", err)
 	}
-	system := System{
+	system := &System{
 		Current: false,
 		Label:   label,
 		Model:   model,
 		Brand:   brand,
 		Actions: defaultSystemActions,
 	}
-	return &system, nil
+	if currentSystem != nil && isCurrentSystem(currentSystem, system) {
```

Comment:
maybe the argument to the helper should be switched for clarity:

isCurrentSystem(system, currentSystem) ?